cfme/infrastructure/provider/__init__.py

```diff
@@ -108,15 +120,107 @@
 mon_btn = partial(tb.select, 'Monitoring')
 
 
-class InfraProviderTimelinesView(TimelinesView, BaseLoggedInPage):
+class InfraProvidersView(BaseLoggedInPage):
+    @property
+    def is_displayed(self):
+        return all((self.logged_in_as_current_user,
+                    self.navigation.currently_selected == ['Compute',
+                                                           'Infrastructure', 'Providers'],
+                    match_page(summary='Infrastructure Providers')))
+
+    @View.nested
+    class toolbar(ProviderToolBar):  # NOQA
+        pass
+
+    @View.nested
+    class sidebar(ProviderSideBar):  # NOQA
+        pass
+
+    @View.nested
+    class entities(ProviderEntities):  # NOQA
+        pass
+
+    @View.nested
+    class paginator(PaginationPane):  # NOQA
+        pass
+
+
+class InfraProviderDetailsView(BaseLoggedInPage):
+    @View.nested
+    class toolbar(DetailsProviderToolBar):  # NOQA
+        pass
+
+    title = Text('//div[@id="main-content"]//h1')
+    breadcrumb = BreadCrumb(locator='//ol[@class="breadcrumb"]')
+
+    properties = SummaryTable(title="Properties")
+    status = SummaryTable(title="Status")
+    relationships = SummaryTable(title="Relationships")
+    overview = SummaryTable(title="Overview")
+    smart_management = SummaryTable(title="Smart Management")
+
+
+class InfraProvidersDiscoverView(BaseLoggedInPage):
+    vcenter = Checkbox('discover_type_virtualcenter')
+    mscvmm = Checkbox('discover_type_scvmm')
+    rhevm = Checkbox('discover_type_rhevm')
+
+    from_ip1 = Input('from_first')
+    from_ip2 = Input('from_second')
+    from_ip3 = Input('from_third')
+    from_ip4 = Input('from_fourth')
+    to_ip4 = Input('to_fourth')
 
+    start = Button('Start')
+    cancel = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        return all((self.logged_in_as_current_user,
+                    self.navigation.currently_selected == ['Compute',
+                                                           'Infrastructure', 'Providers'],
+                    match_page(summary='Infrastructure Providers Discovery')))
+
+
+class InfraProvidersAddView(BaseLoggedInPage):
+    name = Input('name')
+    type = Dropdown('emstype')
+
+    add = Button('Add')
+    cancel = Button('Cancel')
+
+    # todo: rest of entities should be added according to provider type
```

Comment:
Check out PR #4388, hopefully will get merged first but there was a form change in 5.8 if you're looking at the current provisioning_form.